workflow-bot-app/src/main/java/com/symphony/bdk/workflow/WorkflowValidator.java

```diff
@@ -0,0 +1,55 @@
+package com.symphony.bdk.workflow;
+
+import com.symphony.bdk.workflow.engine.WorkflowDirectGraph;
+import com.symphony.bdk.workflow.swadl.exception.ActivityNotFoundException;
+import com.symphony.bdk.workflow.swadl.exception.InvalidActivityException;
+import com.symphony.bdk.workflow.swadl.v1.Event;
+import com.symphony.bdk.workflow.swadl.v1.EventWithTimeout;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+import com.symphony.bdk.workflow.swadl.v1.activity.BaseActivity;
+
+import lombok.experimental.UtilityClass;
+import org.apache.commons.lang3.StringUtils;
+
+@UtilityClass
+public class WorkflowValidator {
+
+  private boolean isConditional(BaseActivity activity) {
+    return activity.getIfCondition() != null || (activity.getOn() != null
+        && activity.getOn().getActivityCompleted() != null
+        && activity.getOn().getActivityCompleted().getIfCondition() != null);
+  }
+
+  public static void validateFirstActivity(BaseActivity activity, Event event, String workflowId) {
+    if (isConditional(activity)) {
+      throw new InvalidActivityException(workflowId,
+          String.format("Starting activity %s cannot have a conditional branching", activity.getId()));
+    }
+    if (event instanceof EventWithTimeout && StringUtils.isNotEmpty(((EventWithTimeout) event).getTimeout())) {
+      throw new InvalidActivityException(workflowId,
+          String.format("Workflow's starting activity %s should not have timeout",
+              activity.getId()));
+    }
+    if (event.getActivityCompleted() != null || event.getActivityFailed() != null
```

Comment:
Is this true when the first activity has on:activity-completed inside a one-of event? Because it should be ok in case we want to loop back to the first activity, since there is another event in one-of list

Reply:
yes, i know, that s why i did not check them in event.getOneOf() list here. However this reminds me that the event itself might already be an event inside the OneOf list.